drivers/espi/espi_emul.c

```diff
@@ -19,168 +19,472 @@ LOG_MODULE_REGISTER(espi_emul_ctlr);
 #include <zephyr/drivers/emul.h>
 #include <zephyr/drivers/espi.h>
 #include <zephyr/drivers/espi_emul.h>
-#include "espi_utils.h"
+#include <zephyr/drivers/espi/espi_utils.h>
+
+/** Declare the default state of virtual wires */
+const static struct espi_emul_vw_data vw_state_default[] = {
```

Comment:
IIRC, ESPI defines a specific set of named vwires, and a system can use them or not use them. (I don't think this is all of them?) I have a doc to emulate VBUS in a way that is devicetree configurable. But vwires (ESPI) are unrelated to VBUS (USB).

If we were to configure how we use vwires based on devicetree, I would imagine it would have something to do with our GPIO config, like if we have a real SLP_S3, we wouldn't use the SLP_S3 vwire. But that's out of scope here.

Reply:
Yes, the eSPI specification defines exactly what virtual-wires exist. There is no way to define custom ones.
It is due to the protocol, since the information about level change on vwire can't be sent separately, the messages for one vwire includes also some other vwires, but marked as "invalid data". The specification describes exactly which vwires are sent with which other.